django/db/backends/base/schema.py

```diff
@@ -321,7 +323,12 @@ def delete_model(self, model):
 
     def add_index(self, model, index):
         """Add an index on a model."""
-        self.execute(index.create_sql(model, self))
+        try:
+            sql = index.create_sql(model, self)
+        except ExpressionIndexNotSupported as e:
+            warnings.warn(str(e), RuntimeWarning)
```

Comment:
While it's an error on platforms w/o support for expression indexes to create one, the presence of an index is often not required. Thus, I'm raising warning instead of blowing up. That's the only way I can imagine for 3rd party packages using expression indexes to use them while supporting all databases.